File: frontend/src/app/models/user-registration.dto.ts
export enum Gender {
  MALE = 'MALE',
  FEMALE = 'FEMALE',
  OTHER = 'OTHER'
}

export interface UserRegistrationDto {
  userName: string;
  surName: string;
  password: string;
  emailId: string;
  age: number;
  gender: Gender;
  height: number;
  weight: number;
  createdAt?: Date;
  updatedAt?: Date;
} 

File: backend/api-gateway-server/src/main/java/com/myportfolio_apigateway/authentication/dto/UserRegistrationDto.java
package com.myportfolio_apigateway.authentication.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserRegistrationDto {

    private String userName;
    private String surName;
    private String password;
    private String emailId;
    private int age;
    private Gender gender;
    private double height;
    private double weight;

    private Date createdAt;
    private Date updatedAt;
}
